rp2040-hal/src/pwm/mod.rs

```diff
@@ -373,6 +373,147 @@ where
     }
 }
 
+/// Pwm Interrupt Control block
+pub struct Interrupt {}
+
+impl Interrupt {
+    /// Enable the interrupt for the indicated channel.
+    pub fn enable_interrupt<S: SliceId, M: SliceMode + ValidSliceMode<S>>(
+        &mut self,
+        _: &Slice<S, M>,
+    ) {
+        match S::DYN {
+            DynSliceId { num } if num == 0 => unsafe {
+                (*pac::PWM::ptr()).inte.write(|w| w.ch0().set_bit())
```

Comment:
This will clear all other bits in the register. The correct method would be
```suggestion
                (*pac::PWM::ptr()).inte.modify(|_, w| w.ch0().set_bit())
```
In practice I'd like to see this using the atomic register access instead. This way parts of the interrupt could be sent between threads.